### PR TITLE
:book: Add documentation on how to support multi-tenancy with webhooks

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
/assign @ncdc @detiber @fabriziopandini 

/cc @yastij @CecileRobertMichon @alexeldeib @justinsb @wfernandes @dlipovetsky @randomvariable @jichenjc

### CODE REVIEW

**Comment by fabriziopandini** [comment]
LGTM to me

File: docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md [hunk 1]
```diff
@@ -154,3 +154,176 @@ unique keys to `failureDomainSpec`s as well as respecting a set `Machine.Spec.Fa
...
+      ```
+
+- In `config/`
+  - **Create**
+    - **kustomization.yaml**: This file is going to function as the new entrypoint to run `kustomize build`.
+      ```yaml
+      namePrefix: capi-{{PROVIDER_NAME}}-{{PROVIDER_TYPE}}-
+
+      commonLabels:
+        cluster.x-k8s.io/provider: "{{PROVIDER_TYPE}}-{{PROVIDER_NAME}}"
```
Comment (vincepri) [inline docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md]:
```suggestion
        cluster.x-k8s.io/provider: "{{PROVIDER_NAME}}-{{PROVIDER_TYPE}}"
```

File: docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md [hunk 2]
```diff
@@ -154,3 +154,187 @@ unique keys to `failureDomainSpec`s as well as respecting a set `Machine.Spec.Fa
...
+      - kind: Certificate
+        group: cert-manager.io
+        path: spec/secretName
+      ```
+
+- In `config/`
+  - **Create**
+    - **kustomization.yaml**: This file is going to function as the new entrypoint to run `kustomize build`.
+      `PROVIDER_NAME` is the name of your provider, e.g. `aws`.
+      `PROVIDER_NAME` is the type of your provider, e.g. `control-plane`, `bootstrap, `infrastructure`.
```
Comment (detiber) [inline docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md]:
```suggestion
      `PROVIDER_TYPE` is the type of your provider, e.g. `control-plane`, `bootstrap`, `infrastructure`.
```

File: docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md [hunk 3]
```diff
@@ -154,3 +154,187 @@ unique keys to `failureDomainSpec`s as well as respecting a set `Machine.Spec.Fa
...
+      - kind: Certificate
+        group: cert-manager.io
+        path: spec/secretName
+      ```
+
+- In `config/`
+  - **Create**
+    - **kustomization.yaml**: This file is going to function as the new entrypoint to run `kustomize build`.
+      `PROVIDER_NAME` is the name of your provider, e.g. `aws`.
+      `PROVIDER_NAME` is the type of your provider, e.g. `control-plane`, `bootstrap`, `infrastructure`.
```
Comment (detiber) [inline docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md]:
```suggestion
      `PROVIDER_TYPE` is the type of your provider, e.g. `control-plane`, `bootstrap`, `infrastructure`.
```

**Comment by detiber** [comment]
/lgtm
/hold cancel
We can always iterate later if needed.

**Pull request merged**